### ISSUE TITLE
--coverage breaks HaxeFlixel project

### ISSUE THREAD

**Issue by nightblade9** [massiveinteractive/MassiveUnit#138]
**TL;DR:** Running `haxelib run munit test` works, but running `haxelib run munit test --coverage` breaks with an odd-looking HaxeFlixel error (` Class<flixel.FlxG> has no field keys`).

[Here's a specific version](https://github.com/nightblade9/eman-adventures/tree/b19ecf7dca79e798fb7534fa47e235f1db9a6370) of a project I'm working on. The project is a mix of HaxeFlixel and non-HaxeFlixel code, while the tests don't touch HaxeFlixel. Thus, when I run `haxelib run munit test`, it compiles and runs all the tests; when I run `haxelib run munit test --coverage`, it fails with this error:

```
Massive Unit - Copyright 2017 Massive Interactive. Version 2.1.2
   haxe -main TestMain -lib munit -lib polygonal-core -lib haxesharp -lib helix -lib flixel -cp src -cp test --remap flash:openfl -neko build/neko_test.n -lib mcover -D MCOVER --macro mcover.MCover.coverage([''],['src'],[''])
HaxeWrapper.hx:73: /home/travis/haxe/lib/helix/git/src/helix/core/HelixState.hx:42: characters 15-24 : Class<flixel.FlxG> has no field keys
HaxeWrapper.hx:73: /home/travis/haxe/lib/helix/git/src/helix/core/HelixState.hx:47: characters 15-24 : Class<flixel.FlxG> has no field keys
Error: Error compiling hxml for neko
Target neko /home/travis/build/nightblade9/eman-adventures/build/neko_test.n
The command "haxelib run munit test -coverage -result-exit-code" exited with 1.
```

The difference `--coverage` makes is the addition of `-lib mcover -D MCOVER --macro mcover.MCover.coverage([''],['src'],[''])`, which I suspect is causing the problem.

[Here's a build that fails](https://travis-ci.org/nightblade9/eman-adventures/builds/287985687) with this error.

I'm pretty sure this is not a HaxeFlixel bug, or a bug in my code, because I can compile and run the game, and compile and run the tests (without `--coverage`). It might be a bug in my config somewhere.

`test.hxml`:

```
## Neko
-main TestMain
-lib munit
-lib polygonal-core
-lib haxesharp
-lib helix
-lib flixel

-cp src
-cp test

--remap flash:openfl

-neko build/neko_test.n
```

I'm currently working around this by removing `--coverage` from my test execution, which works, but isn't ideal.

**Comment by Gama11** [comment]
>Class<flixel.FlxG> has no field keys

[`FLX_KEYBOARD`](https://github.com/HaxeFlixel/flixel/blob/4.3.0/flixel/FlxG.hx#L211-L217) is likely not defined, because Flixel's initialization macro isn't being run. Try adding this to your hxml:

```
--macro flixel.system.macros.FlxDefines.run()
```

**Comment by nightblade9** [comment]
@Gama11 that seems to work. The compiler error disappears, but a new one appears: 

```
HaxeWrapper.hx:73: D:\HaxeToolkit\haxe\lib\flixel/4,3,0/flixel/input/gamepad/FlxGamepadManager.hx:419: characters 46-59 : Type not found : Joy
stickEvent
```

Also, it's not clear why I don't need this when running the tests without `--coverage`.

**Comment by Gama11** [comment]
Sounds like `FLX_GAMEINPUT_API` isn't being defined by the init macro.

https://github.com/HaxeFlixel/flixel/blob/4.3.0/flixel/system/macros/FlxDefines.hx#L161

Maybe try adding `-D openfl_next`?

**Comment by nightblade9** [comment]
@Gama11 looks like that fixed it. Thanks!

Does anyone have any idea why this is required for `--coverage` but works fine without it?

**Comment by bendmorris** [comment]
>Does anyone have any idea why this is required for --coverage but works fine without it?

`--coverage` runs a macro which includes files in the package you're measuring coverage for. They may not have been included when just running the tests.
